### PR TITLE
[WIP] Fixing the powernet!  Now with GRAPHS!  

### PR DESCRIPTION
**Pull request by WarlockD**
## About The Pull Request
Yes Yes, Powernet sucks.  So giving it an update.  Put in a datum that handles graphs by itself so we can use it for other graph needs (aka pipenet, plumming etc) so that we never get those dangling nodes ever again...hah right

## Why It's Good For The Game
To stop people complaing about powernet

## Changelog
:cl:
refactor: refactored powernet
/:cl:

### CODE REVIEW

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -1,17 +1,21 @@
 //Use this only for things that aren't a subtype of obj/machinery/power
 //For things that are, override "should_have_node()" on them
-GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/grille)))
+▬GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/grille)))
```
Comment (bobbahbrown) [inline code/modules/power/cable.dm:3]:
```suggestion
GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/grille)))
```

Comment (WarlockD) [reply code/modules/power/cable.dm]: oops.  stray finger hit there

**Comment by LemonInTheDark** [comment]
Resolves #51059?

**Comment by WarlockD** [comment]
I hate you all.  That new line merged screwed my branches, I figured out how I am going to do it atleast so watch out for my next pull on this.

**Comment by LemonInTheDark** [comment]
holding out for a hero.....

**Pull request closed without merging**